python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -267,6 +267,39 @@ def test_actor_method_bind_same_constant(ray_start_regular):
     compiled_dag.teardown()
 
 
+def test_actor_method_bind_same_arg(ray_start_regular):
+    actor = Actor.remote(1)
+    with InputNode() as inp:
+        one = actor.echo.bind(inp)
+        # Test binding the same arg (one) to the same method
+        # (double_and_inc) of same actor multiple times:
+        # execution should not hang.
+        output1 = actor.double_and_inc.bind(one)
+        output2 = actor.double_and_inc.bind(one)
+        dag = MultiOutputNode([output1, output2])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(1)
+    result = ray.get(ref)
+    assert result == [3, 7]
+    compiled_dag.teardown()
+
+
+def test_actor_method_bind_same_input(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # Test binding input node to the same method
+        # of same actor multiple times: execution
+        # should not hang.
+        output1 = actor.inc.bind(inp)
+        output2 = actor.inc.bind(inp)
+        dag = MultiOutputNode([output1, output2])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(1)
+    result = ray.get(ref)
+    assert result == [1, 2]
+    compiled_dag.teardown()
+
+
```

Comment:
can you also add this case? Want to test input is used by 2 workers + used twice by 1 worker at the same time
```
with InputNode as inp:
    out = s1.fwd(inp)
    out2 = s2.fwd(inp, out)
    out3 = s1.fwd_2(out2)
```

Reply:
hmm, can you clarify the example?
`fwd` is just called by s1 once, and s2 once? How many args does `fwd` take? 

Did you mean something like this?
```
    with InputNode() as inp:
        output1 = a1.inc.bind(inp)
        output2 = a1.inc.bind(inp)
        output3 = a2.inc.bind(inp)
        dag = MultiOutputNode([output1, output2, output3])
```